### PR TITLE
add regression test for scala/bug#11106

### LINKED ISSUE #11106: Iterator's laziness is incomplete
```
sbt:satotakeru> set scalaVersion := "2.13.0-M4"
[info] Defining scalaVersion
[info] The new value will be used by Compile / dependencyGraphCrossProjectId, Compile / moduleGraph and 29 others.
[info]  Run `last` for details.
[info] Reapplying settings...
[info] Set current project to satotakeru (in build file:/Users/satotakeru/)
sbt:satotakeru> console
[info] Starting scala interpreter...
Welcome to Scala 2.13.0-M4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> var i = 0; Iterator.continually(0).map(_ => {i += 1; i}).withFilter(_ < 10).take(3).foreach(_ => ()); println(i)
4
i: Int = 4

scala> :q

[success] Total time: 13 s, completed Aug 24, 2018 1:05:35 PM
sbt:satotakeru> set scalaVersion := "2.13.0-M3"
sbt:satotakeru> console
[info] Non-compiled module 'compiler-bridge_2.13.0-M2' for Scala 2.13.0-M3. Compiling...
[info]   Compilation completed in 13.512s.
[info] Starting scala interpreter...
Welcome to Scala 2.13.0-M3 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_172).
Type in expressions for evaluation. Or try :help.

scala> var i = 0; Iterator.continually(0).map(_ => {i += 1; i}).withFilter(_ < 10).take(3).foreach(_ => ()); println(i)
3
i: Int = 3
```

Chaining `withFilter` after `map` seems to be evaluating `map` beyond the number passed to `take`.

### PR DESCRIPTION
**Pull request by ghost**
Add regression test for a already fixed bug (scala/bug#10883 with the PR #6784).

Fixes scala/bug#11106.

### CODE REVIEW

File: test/junit/scala/collection/immutable/LazyListTest.scala [hunk 1]
```diff
@@ -1000,4 +1000,15 @@ class LazyListTest {
       assertEquals(s"mkString 3 $i", goal(3,i), precyc(3,i).mkString)
     }
   }
+
+  @Test
+  def t11106: Unit = {
```
Comment (NthPortal) [inline test/junit/scala/collection/immutable/LazyListTest.scala]: I don't think this test belongs in `LazyListTest.scala` - it doesn't really have anything to do with `LazyList`.

**Comment by ghost** [comment]
Sure, sorry for that.

It's more a **Iterator** test.

**Comment by NthPortal** [comment]
@rfossati no worries!

**Review by NthPortal — APPROVED** [review]
LGTM

I'm not sure if the `withFilter` in the snippet in bug was an important part of the bug or not (since I'm not sure exactly where the bug was fixed)

**Comment by SethTisue** [comment]
whoops, whoever submitted this has vanished from GitHub. 

would someone like to resubmit this...?

**Pull request closed without merging**